client/command/filesystem/download.go

```diff
@@ -114,11 +153,23 @@ func HandleDownloadResponse(download *sliverpb.Download, ctx *grumble.Context, c
 			return
 		}
 		if err == nil && fi.IsDir() {
+			if download.IsDir {
+				// Come up with a good file name - filters might make the filename ugly
+				fileName = fmt.Sprintf("%s_download_%s_%d.tar.gz", con.GetActiveSessionConfig().Name, prettifyDownloadName(remotePath), time.Now().Unix())
```

Comment:
you need to call `filepath.Base()` on the file name components (except for the timestamp), the session name can potentially be set by other users and while we restrict the chars it could lead to path traversal.

Reply:
Gotcha. I did not realize the session name could be set by the user. I thought that was system generated and was set once. I was not as worried about the result of `prettifyDownloadName` because it produces a string with alphanumeric characters and underscores. However, it does not hurt to call `filepath.Base()` on it as well. Good catch - thank you!